x-pack/plugins/security_solution/common/api/detection_engine/prebuilt_rules/model/diff/three_way_diff/three_way_diff.ts

```diff
@@ -19,6 +19,12 @@ import type { ThreeWayMergeOutcome } from './three_way_merge_outcome';
 export const MissingVersion = Symbol('MissingVersion');
 export type MissingVersion = typeof MissingVersion;
 
+export enum ThreeWayDiffConflictResolutionResult {
+  NO_CONFLICT = 'NO_CONFLICT',
+  SOLVABLE_CONFLICT = 'SOLVABLE_CONFLICT',
+  NON_SOLVABLE_CONFLICT = 'NON_SOLVABLE_CONFLICT',
+}
```

Comment:
For the sake of brevity, could we please do this:

```suggestion
export enum ThreeWayDiffConflict {
  NONE = 'NONE',
  SOLVABLE = 'SOLVABLE',
  NON_SOLVABLE = 'NON_SOLVABLE',
}
```

Also:

- I think it deserves its own file in consistency with `ThreeWayDiffOutcome` and `ThreeWayMergeOutcome`.
- Please add JSDoc comments for the whole enum and its values. It would be great to refer to the corresponding sections of the RFC.

Reply:
This was initially so and I expanded for clarity after getting some [feedback.](https://github.com/elastic/kibana/pull/187770#discussion_r1679916323)

But I'll extract it to its own file and add JSDoc comments explaining this and that should be enough to make the enum clear enough to anyone using it.